src/GovUk.Education.ExploreEducationStatistics.Data.Services/ViewModels/LocationViewModelBuilder.cs

```diff
@@ -0,0 +1,123 @@
+﻿#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels.Meta;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels;
+
+public class LocationViewModelBuilder
+{
+    public static Dictionary<GeographicLevel, List<LocationAttributeViewModel>> BuildLocationAttributeViewModels(
+        IList<Location> locations,
+        Dictionary<GeographicLevel, List<string>>? hierarchies,
+        Dictionary<GeographicLevel, Dictionary<string, GeoJson>>? geoJson = null)
+    {
+        var locationAttributes = locations.GetLocationAttributesHierarchical(hierarchies);
+        return locationAttributes.ToDictionary(
+            pair => pair.Key,
```

Comment:
Might be tempted to call "pair" something that hints at what is contained in the pair, like "codeAndLocationAttributes", but not massively fussed.

Reply:
Thanks have renamed it `levelAndLocationAttributes` although I think it would be really nice if the language allowed something like

```
.ToDictionary(
    (level, _) => level,
    (level, locationAttributes) => DoSomething(level, locationAttributes)
)